### PR TITLE
cache configuration: Add gsettings config backend support.

### PR DESCRIPTION
**Pull request by sunweaver**

### CODE REVIEW

File: lib/cacheconfiguration.h [hunk 1]
```diff
@@ -41,6 +41,9 @@
 #ifdef HAS_MLITE
 #include <mgconfitem.h>
 #endif
+#ifdef HAS_GSETTINGS
```
Comment (pvuorela) [inline lib/cacheconfiguration.h]: Suppose QGSETTINGS would be more precise naming for this. On GSettings I first assumed this was about glib.

File: lib/cacheconfiguration.h [hunk 2]
```diff
@@ -72,12 +75,22 @@ class CONTACTCACHE_EXPORT CacheConfiguration : public QObject
...
+#endif // HAS_GSETTINGS
 
+#ifdef HAS_MLITE
 private slots:
     void onDisplayLabelOrderChanged();
     void onSortPropertyChanged();
     void onGroupPropertyChanged();
 #endif
+#ifdef HAS_GSETTINGS
+private slots:
```
Comment (pvuorela) [inline lib/cacheconfiguration.h]: Minor detail but could take the private slots to common part? Alternatively if the new connect() syntax with method pointers was used, there wouldn't be need for declaring such a thing.

File: lib/cacheconfiguration.h (hunk 1 above)
Comment (sunweaver) [reply lib/cacheconfiguration.h]: It is about GSETTINGS from glib! (via gsettings-qt).

Comment (sunweaver) [reply lib/cacheconfiguration.h]: Note that Ubuntu Touch / Lomiri is a Qt based shell but using the glib2.0 gsettings backend for storing configuration data.

Comment (pvuorela) [reply lib/cacheconfiguration.h]: Sure but the component needed here is not gsettings directly. For comparison mlite part is really a wrapper to dconf, but it's not HAS_DCONF. (And gsettings itself is something like a dconf wrapper  too)

**Comment by sunweaver** [comment]
@pvuorela Please recheck, I have rebased this PR against 0.3.30 now and followed your pimplify code changes.

File: lib/cacheconfiguration.cpp [hunk 3]
```diff
@@ -135,6 +167,45 @@ void CacheConfigurationPrivate::onGroupPropertyChanged()
 }
 #endif
 
+#ifdef HAS_QGSETTINGS
+void CacheConfigurationPrivate::onConfigPropertyChanged(const QString &key) {
```
Comment (pvuorela) [inline lib/cacheconfiguration.cpp:171]: Nitpick: by convention the method starting braces on their own lines but I can adjust.

**Review by pvuorela — APPROVED** [review]

**Pull request merged**